### ISSUE TITLE
Skeleton File or Default YAML Block

### ISSUE THREAD

**Issue by Fleckenteufel** [typora/typora-issues#863]
It would be great to be able to define a default YAML block that is entered in every file that is created. This would allow to define an image folder or a root URL:
```
typora-copy-images-to: .images
typora-root-url: ./
```
Alternatively, this could be done by defining a skeleton file that is used instead of an empty file when creating a new one.

Related to: #274, #423

**Comment by dejan-stankovic** [comment]
great suggestion! I would also like to be able hide YAML Block as it is annoying/distracting. it could be that i just didn't find a way how to hide it...

**Comment by asins** [comment]
Has this function been added? Every time you paste a picture in a new document, you have to add YAML information first, which interrupts the user's operation.

这个功能有加入吗？每次在新文档中粘贴图片都要先添加YAML信息，这打断了用户的操作。

**Comment by GandalfSaxe** [comment]
I agree; I really only use the YAML header for providing a path for any pasted images, which is required. It'd be nice to have a default location for these, and a possible way to do this would be a default YAML that always applies (bonus points if this can be hidden).

Even better solution would be support for [textbundle](http://textbundle.org), which is the most elegant solution to keeping plaintext MD files and their associated image assets together.

**Comment by sheldonhull** [comment]
I came looking for a way to setup something similar to prose config with typora. I'd love to setup my default front matter for jekyll posts as well as default image paths. Since I use `/assets/img/imagename`, having this as front-matter for all my posts in a directory would be useful. 

Perhaps use the same concept as a config in jekyll or git ignore. Allow a .typoraconfig in a directory, or subdirectory to override with default values for front matter. This would give a lot of flexibility for using Typora for blogging through Jekyll, and other static tools, as well as for other purposes.

**Comment by vassudanagunta** [comment]
@abnerlee It would be almost trivial for Typora to add two new menu items (or two buttons in the preferences panel):

- **Edit New File Template** opens `markdown-template.md` in the editor. This file is stored in the same user space as custom CSS, and is used as the template whenever the user chooses `File:New`.
- **Edit New YAML Front Matter Template** opens `yaml-template.md` in the editor.  This file is stored in the same user space as custom CSS, and is used as the template whenever the user chooses `Paragraph:YAML Front Matter`.

Keeping them as separate options provides the most flexibility, as only supporting the first option doesn't help people who have to add front matter to existing files. 

I often have to add a category and tags to my documents and have to manually type the following every time: 

```
category: 
tags: []
```

**Comment by gensmusic** [comment]
I really need a global setting for ` typora-root-url`, I always work on different platforms like osx, linux and windows, it's fustrating to set the `typora-root-url` as the folder cann't be the same on different platforms

**Comment by abnerlee** [comment]
Use `duplicate` in file panel would be a workaround in new version

**Comment by Sandaidev** [comment]
Please, like @gensmusic, I use Windows alongside Linux and it's really frustrating to use !

**Comment by christophcemper** [comment]
We need the same kind of templating function for YAML

**Comment by TigersWay** [comment]
> Perhaps use the same concept as a config in jekyll or git ignore. Allow a .typoraconfig in a directory, or subdirectory to override with default values for front matter. This would give a lot of flexibility for using Typora for blogging through Jekyll, and other static tools, as well as for other purposes.

That would be Soooo great!!!

**Comment by sheldonhull** [comment]
⬆️  bump. Would really like to have either:

- Default config in a project directory that sets this 1x so I don't have to include in the front matter of every 200+ files
- Ability to set as a default front matter value that automatically gets set if doesn't exist.

**Comment by 0x413x4** [comment]
+1

I would really like to see a feature akin to the templates in Obsidian, whereby you click on an icon, select a template, and it copies the content of the template in the current document. Bonus point if we can have *very basic* placeholders, such as {{filename}} or {{date}}.

**Comment by 1999FordFocus** [comment]
I use Typora for local notes stuff, and I believe a global root image URL setting is better than a template or manual copy.

Please sir! Really really need this.

**Comment by sheldonhull** [comment]
Bump. Still something I'd like to see. Right now my repo front matter is covered typora values to make sure it can resolve images for blogging. I'd love to have a config file in my repo root or some type of way to set this besides per document.

**Comment by AyiStar** [comment]
Bump. Need this feature, too. Is there any updates on the development?

**Comment by sospiess** [comment]
YES! This would be a really useful feature. 
To me, the proposal by @0x413x4 would have the most benefit. I don't need one template automatically applied all the time, but have several _types of documents_ I use frequently and would like some consitency. For instance, one template for meeting minutes, another one for user stories, etc..

Right now, I first need navigate to and open a template (which probably isn't in the currently opended folder and thus easily selectable), and then SAVE AS it in the correct location (which makes it dangerous to inadvertenly overwrite the template with autosave turned on, when I start jotting away meeting attendees' remarks before I explicitly save the document.

Being able to store "read-only" document templates somewhere, which get opened as new files/copies in the background would be just wonderful. Those templates could easily be prepared with or without a yaml frontmatter by the user.
Finally, one of those templates could also be set as the default for all "new" documents, if so desired by a user.

**Comment by myob-hankun** [comment]
Hi Team,
Do we have a plan to do template feature?
It's important
